probe-rs/src/vendor/nxp/sequences/nxp_armv6m.rs

```diff
@@ -0,0 +1,174 @@
+            interface.write_word_32(Dhcsr::get_mmio_address(), dhcsr.into())?;
+            let start = Instant::now();
+            while start.elapsed() < Duration::from_millis(100) {
+                if Dhcsr(interface.read_word_32(Dhcsr::get_mmio_address())?).s_halt() {
+                    break;
+                };
+            }
+        }
+
+        tracing::trace!("force_core_halt exit");
+        Ok(())
+    }
+}
+
+impl ArmDebugSequence for LPC80x {
+    fn reset_catch_set(
+        &self,
+        interface: &mut dyn ArmMemoryInterface,
+        _core_type: crate::CoreType,
+        _debug_base: Option<u64>,
+    ) -> Result<(), ArmError> {
+        tracing::trace!("reset_catch_set enter");
+
+        // Disable Reset Vector Catch in DEMCR
+        let mut demcr = Demcr(interface.read_word_32(Demcr::get_mmio_address())?);
+        demcr.set_vc_corereset(false);
+        interface.write_word_32(Demcr::get_mmio_address(), demcr.into())?;
+
+        // Map Flash to Vectors
+        interface.write_word_32(0x4004_8000, 0x0000_0002)?;
+
+        // Read reset vector from Flash
+        let reset_vector = interface.read_word_32(0x0000_0004)?;
+        tracing::trace!("Reset Vector is address 0x{:08x}", reset_vector);
+
+        LPC80x::set_hw_breakpoint(interface, 0, reset_vector)?;
+
+        // Clear the status bits by reading from DHCSR
+        let _ = interface.read_word_32(Dhcsr::get_mmio_address())?;
+        tracing::trace!("reset_catch_set exit");
+
+        Ok(())
+    }
+
+    fn reset_catch_clear(
+        &self,
+        interface: &mut dyn ArmMemoryInterface,
+        _core_type: crate::CoreType,
+        _debug_base: Option<u64>,
+    ) -> Result<(), ArmError> {
+        tracing::trace!("reset_catch_clear enter");
+
+        // Disable Reset Vector Catch in DEMCR
+        let mut demcr = Demcr(interface.read_word_32(Demcr::get_mmio_address())?);
+        demcr.set_vc_corereset(false);
+        interface.write_word_32(Demcr::get_mmio_address(), demcr.into())?;
+
+        LPC80x::clear_hw_breakpoint(interface, 0)?;
+
+        tracing::trace!("reset_catch_clear exit");
+        Ok(())
+    }
+
+    fn reset_system(
+        &self,
+        interface: &mut dyn ArmMemoryInterface,
+        _core_type: crate::CoreType,
+        _debug_base: Option<u64>,
+    ) -> Result<(), ArmError> {
+        tracing::trace!("reset_system enter");
+
+        // Execute VECTRESET via AIRCR, ignore errors.
+        let mut aircr = Aircr(0);
+        aircr.vectkey();
+        aircr.set_sysresetreq(true);
+        let _ = interface.write_32(Aircr::get_mmio_address(), &[aircr.0]);
+
+        let start = Instant::now();
+        while start.elapsed() < Duration::from_millis(100) {
+            // ignore read errors while resetting
+            if let Ok(dhcr) = interface.read_word_32(Dhcsr::get_mmio_address()) {
+                if Dhcsr(dhcr).s_halt() {
+                    // break from loop if we're in debug state
+                    break;
```

Comment:
If this happens, couldn't we just exit this function? `force_core_halt` seems to just check the same condition again, so it seems pointless if we know already that the core stopped.

Reply:
Yes, that is true. I did notice this while porting but I was restricting myself to doing exactly what the original sequence did.
Unfortunately forgot to come back to it afterwards. I've tested this modification and it still works correctly, and added it to this PR in 4fb9f7f